chaossaltstack/machine/actions.py

```diff
@@ -602,6 +603,94 @@ def network_latency(instance_ids: List[str] = None,
         )
 
 
+def kill_processes(instance_ids: List[str] = None,
+                   execution_duration: str = "60",
+                   process_name: str = None,
+                   configuration: Configuration = None,
+                   secrets: Secrets = None):
+    """
+    Increases the response time of the virtual machine.
+
+    Parameters
+    ----------
+    instance_ids : List[str]
+        Filter the virtual machines. If the filter is omitted all machines in
+        the subscription will be selected as potential chaos candidates.
+    execution_duration : str, optional default to 1 second
+        This is not technically not useful as the process usually is killed
+        without and delay, however you can set more seconds here to let the
+        thread wait for more time to extend your experiment execution in case
+        you need to watch more on the observation metrics.
+    process_name : str
+        Name of the process to be killed
+    configuration : Configuration
+        Chaostoolkit Configuration
+    secrets : Secrets
+        Chaostoolkit Secrets
+    """
+    logger.debug(
+        "Start network_latency: configuration='{}', instance_ids='{}'".format(
+            configuration, instance_ids))
+
+    try:
+        client = saltstack_api_client(secrets)
+        machines = client.get_grains_get(instance_ids, 'kernel')
+
+        param = dict()
+        param["duration"] = execution_duration
+        param["param"] = process_name
+
+        jids = dict()
+
+        if len(machines) <= 0:
+            FailedActivity(
+                "Cannot find any machines {}".format(instance_ids))
+
+        for k, v in machines.items():
+            name = k
+            os_type = v
+            param["instance_id"] = k
+            script_content = __construct_script_content__(
+                KILL_PROCESSES, os_type, param)
+
+            # Do async cmd and get jid
+            logger.debug("network_latency of machine: {}".format(name))
+            salt_method = 'cmd.run'
+            jid = client.async_run_cmd(name, salt_method, script_content)
+            jids[k] = jid
+        logger.debug(json.dumps(jids))
```

Comment:
It's good practice to add a message with a log, so one can understand the context a bit more :)

Reply:
As I find there are too many duplicates in each experiment acton, I will abstract the workflow to one parameterized internal one.